test/request.js

```diff
@@ -246,3 +246,26 @@ test('DispatchOptions#reset', scope => {
     })
   })
 })
+
+test('should throw timeout immediately', scope => {
+  scope.plan(1)
+
+  scope.test('throw headersTimeoutError', t => {
+    t.plan(1)
+
+    const server = createServer((req, res) => {
+      setTimeout(() => {
+        res.end('banana')
```

Comment:
ending the request will likely make this test flaky

Reply:
My intention was to expect the results to be returned after `800ms`
This is to test whether the headerTimeout is triggered successfully.

Reply:
the test will be flaky if the request is being ended, we've had similar issues in the past.